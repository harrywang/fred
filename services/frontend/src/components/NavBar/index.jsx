import React from 'react'
import { Menu } from 'antd'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import styles from './index.module.scss'

const NavBar = props => {
  const { logoutUser, isAuthenticated } = props

  const menu = isAuthenticated('in the nav bar') ? (
    <Menu theme="light" mode="horizontal">
      <Menu.Item key="status">
        <Link to="/status" data-testid="nav-status">
          User Status
        </Link>
      </Menu.Item>
      <Menu.Item key="logout">
        <div
          onClick={() => {
            logoutUser()
          }}
          data-testid="nav-logout"
        >
          Log Out
        </div>
      </Menu.Item>
    </Menu>
  ) : (
    <Menu theme="light" mode="horizontal">
      <Menu.Item key="register">
        <Link to="/register" data-testid="nav-register">
          Register
        </Link>
      </Menu.Item>
      <Menu.Item key="login">
        <Link to="/login" data-testid="nav-login">
          Log In
        </Link>
      </Menu.Item>
    </Menu>
  )

  return (
    <div className={styles.navbar_wrap}>
      <div
        className={styles.navbar_logo}
        onClick={() => {
          window.location.href = '/'
        }}
      >
        Fred
      </div>
      {menu}
    </div>
  )
}

NavBar.propTypes = {
  logoutUser: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.func.isRequired,
}

export default NavBar
