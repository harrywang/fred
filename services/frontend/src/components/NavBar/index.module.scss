.navbar_wrap {
  display: flex;
  width: 100%;
  justify-content: space-between;
}

.navbar_logo {
  font-weight: 700;
  cursor: pointer;
}
