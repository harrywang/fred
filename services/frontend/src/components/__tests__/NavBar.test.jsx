import React from 'react'
import { cleanup, wait } from '@testing-library/react'

import NavBar from '../NavBar'

afterEach(cleanup)

describe('when unauthenticated', () => {
  const props = {
    logoutUser: () => {
      return true
    },
    isAuthenticated: jest.fn().mockImplementation(() => false),
  }

  test('renders the default props', async () => {
    const { getByText, findByTestId } = renderWithRouter(<NavBar {...props} />)

    await wait(() => {
      expect(props.isAuthenticated).toHaveBeenCalledTimes(1)
    })

    expect((await findByTestId('nav-register')).innerHTML).toBe('Register')
    expect((await findByTestId('nav-login')).innerHTML).toBe('Log In')
  })

  test('renders', () => {
    const { asFragment } = renderWithRouter(<NavBar {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('when authenticated', () => {
  const props = {
    title: 'Hello, World!',
    logoutUser: () => {
      return true
    },
    isAuthenticated: jest.fn().mockImplementation(() => true),
  }

  test('renders the default props', async () => {
    const { getByText, findByTestId } = renderWithRouter(<NavBar {...props} />)

    await wait(() => {
      expect(props.isAuthenticated).toHaveBeenCalledTimes(1)
    })

    expect((await findByTestId('nav-status')).innerHTML).toBe('User Status')
    expect((await findByTestId('nav-logout')).innerHTML).toBe('Log Out')
  })

  test('renders', () => {
    const { asFragment } = renderWithRouter(<NavBar {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
