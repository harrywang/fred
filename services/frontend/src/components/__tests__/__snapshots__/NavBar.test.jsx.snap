// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when authenticated renders 1`] = `
<DocumentFragment>
  <div
    class="navbar_wrap"
  >
    <div
      class="navbar_logo"
    >
      Fred
    </div>
    <ul
      class="ant-menu ant-menu-light ant-menu-root ant-menu-horizontal"
      role="menu"
    >
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        <a
          data-testid="nav-status"
          href="/status"
        >
          User Status
        </a>
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        <div
          data-testid="nav-logout"
        >
          Log Out
        </div>
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="visibility: hidden; position: absolute; display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`when unauthenticated renders 1`] = `
<DocumentFragment>
  <div
    class="navbar_wrap"
  >
    <div
      class="navbar_logo"
    >
      Fred
    </div>
    <ul
      class="ant-menu ant-menu-light ant-menu-root ant-menu-horizontal"
      role="menu"
    >
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        <a
          data-testid="nav-register"
          href="/register"
        >
          Register
        </a>
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        <a
          data-testid="nav-login"
          href="/login"
        >
          Log In
        </a>
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="visibility: hidden; position: absolute; display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
